src/interface/blas1_interface.hpp

```diff
@@ -165,16 +151,22 @@ typename sb_handle_t::event_t _sdsdot(
     sb_handle_t &sb_handle, index_t _N, float sb, container_0_t _vx,
     increment_t _incx, container_1_t _vy, increment_t _incy, container_2_t _rs,
     const typename sb_handle_t::event_t &_dependencies) {
-  typename sb_handle_t::event_t dot_event{};
   auto rs = make_vector_view(_rs, static_cast<increment_t>(1),
                              static_cast<index_t>(1));
-
-  dot_event =
-      internal::_dot(sb_handle, _N, _vx, _incx, _vy, _incy, _rs, _dependencies);
-  auto addOp = make_op<ScalarOp, AddOperator>(sb, rs);
-  auto assignOp2 = make_op<Assign>(rs, addOp);
-  auto ret2 = sb_handle.execute(assignOp2, dot_event);
-  return blas::concatenate_vectors(dot_event, ret2);
+  if (!_N) {
+    auto addOp = make_op<ScalarOp, AddOperator>(sb, rs);
```

Comment:
if `_N` is zero this is a copy operation. Did you try to copy directly the value instead of adding and then assigning it? Would make sense to try it?